### ISSUE TITLE
KeyError for missing `loss_sparse` in validation phase of DAZZLE `runner.py`

### ISSUE THREAD

**Issue by Shirshak52** [TuftsBCB/grnvae#1]
In `runner.py` at line 320 within the evaluation block, the code attempts to access `loss_sparse` from the output dictionary returned by the DAZZLE model's `forward()` method, as shown here:
`eval_log['val_loss_sparse'] += out['loss_sparse'].detach().cpu().item()`

However, the DAZZLE model's `forward()` method does not include `loss_sparse` in its returned dictionary. Instead, the sparsity loss (`loss_sparse`) is computed externally in the training loop, e.g.:
```
adj_m = vae.get_adj_()
loss_sparse = torch.norm(adj_m, 1) / n_gene / n_gene
```
Because `loss_sparse` is never part of the model output during validation, this causes a `KeyError`.
